chart/templates/clusterrole.yaml

```diff
@@ -63,10 +63,13 @@ rules:
     resources: ["csistoragecapacities"]
```

Comment:
Since the cluster role is not always available, snapshot sometimes doesn't work like you expect I guess, have you thought about this?

Reply:
Good reminder, thanks!

Yes, I was looking into this, that's remaining to be done, forgot to add it to the TODO list in the PR description.

The original idea was to use the config for enabling volume snapshots to check if the role has to be deployed, but we don't have such config anymore for volume snapshots.

So now I think that we have to always deploy the ClusterRole (also Role), because otherwise volume snapshots will not work. These roles would then always have minimal permissions with just VolumeSnapshot, VolumeSnapshotContent and VolumeSnapshotClass resources being specified.

Reply:
We can also add back some explicit config for this, so users have to enable it explicitly (e.g. like deploying upstream snapshot-controller with private nodes).